### DISCUSSION TITLE
Proposal: Implementing Kriging Interpolation for xarray

### DISCUSSION THREAD

**Discussion by mohseniaref** [pydata/xarray#7926] [Ideas] [answered]
Hello everyone,

I have been using xarray extensively and appreciate its power in handling multi-dimensional array-based data. It's especially useful for gridded geospatial data, such as raster datasets. Currently, xarray supports several interpolation methods like nearest, linear, and cubic.

However, for geospatial data, especially in the field of geostatistics, Kriging interpolation often serves as a more appropriate method. Kriging is a group of geostatistical techniques to interpolate the value of a random field at an unobserved location from observations of its value at nearby locations.

The idea I want to propose here is to implement Kriging interpolation as a part of xarray's interpolation methods.

`kriged = ds.interp(method='kriging', variogram_model='spherical')`

I am eager to hear your thoughts on this proposal and whether you think this could be a useful addition to xarray.

**Comment by TomNicholas** [comment, marked as answer]
Hi @mohseniaref - thank you for this suggestion. This sounds like a useful feature for a number of people, but it also sounds like it would be of interest primarily to a single community (geostatistics). In Xarray we mostly try to include only generic features which are general utility across a number of different scientific fields (e.g. bioscience, fluid dynamics, etc.). 

Instead I suggest you take a look at the [documentation on extending xarray using accessors](https://docs.xarray.dev/en/stable/internals/extending-xarray.html) - you can use that to make a syntax like this work:

```python
kriged = ds.geostats.interp(method='kriging', variogram_model='spherical')
```

Then you might want to look at some of the many geoscience-specific libraries that exist on top of xarray. There is a (non-exhaustive) list on our [ecosystem page](https://docs.xarray.dev/en/stable/internals/extending-xarray.html). One of those might be a better home for your suggestion!

**Reply by mohseniaref** [reply]
Thanks so much for the reply,I still think this interpolation method could be even useful for people outside geoscience

**Comment by dcherian** [comment]
Does [verde](https://github.com/fatiando/verde) do this?

**Reply by mohseniaref** [reply]
I don't think so! @dcherian
